packages/react-ai/src/hooks/useAIGeneration.tsx

```diff
@@ -55,7 +55,10 @@ export function createUseAIGeneration<
     ): Promise<Schema[Key]['returnType']> => {
       const result = await handleGenerate(input);
 
-      return { ...result };
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      const data = (result as any).data as Schema[Key]['returnType'];
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-member-access
+      return { ...data, ...(result as any).errors };
```

Comment:
Can you add comment on why this is needed?

Reply:
Kind of confused what we are doing here, isn't this forcing a return value that doesn't match the shape of `Schema[Key]['returnType']`?

Reply:
Discussed a bit offline, but the return type of the handleGenerate call actually wraps the `Schema[Key]['returnType']` in a Promise with graphqlErrors and some other V6Client related info, the data key of that return type matches the `Schema[Key]['returnType']`.

In terms of adding the errors, the graphql errors do not throw so it would not be caught by the useDataState layer. There is still a chance that the appsync layer could throw an error that would be caught, but we need a way to surface graphql errors to the customer as well

Reply:
the issue this is fixing is that the type didn't match what is actually returned, and the type was the right DX, so this is massaging the actual data to match the type. 

![CleanShot 2024-09-06 at 13 53 15@2x](https://github.com/user-attachments/assets/8fa0e8d2-d321-4732-b9e4-169fb4c001a7)

Reply:
@dbanksdesign `anything.anything` makes me feel crazy. Think the fix there is to return `data` from `handleGenerate`:
```diff
- const result = await handleGenerate(input);
+ const result = await handleGenerate(input).data;
```

Synced with @thaddmt offline abt the `errors` piece since we are moving them up a layer think that change requires us to update the return shape of `updateAIGenerationStateAction`:

```diff
- Promise<Schema[Key]['returnType']>
+ Promise<Schema[Key]['returnType'] & { graphqlErrors: string[] }>
```

Does that make sense?

Reply:
Kinda have concerns abt the underlying types we're consuming from `Schema` here as AFAICT they do not match the shape returned from `handleGenerate` (obviously something we don't own and maybe i'm misunderstanding them)